### PR TITLE
Cody Web: Error handling, especially for rate limits

### LINKED ISSUE #51090: Handle rate limit errors correctly
Since #50925 we have rate limiting for the completions API. We need to make sure that the web client handles this correctly.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #51090

## Test plan

Screenshot 2023-04-27 at 13 16 20

### CODE REVIEW

File: client/cody-shared/src/sourcegraph-api/completions/browserClient.ts [hunk 1]
```diff
@@ -20,14 +20,26 @@ export class SourcegraphBrowserCompletionsClient extends SourcegraphCompletionsC
             headers: Object.fromEntries(headersInstance.entries()),
             body: JSON.stringify(params),
             signal: abort.signal,
+            async onopen(response) {
+                if (!response.ok && response.headers.get('content-type') !== 'text/event-stream') {
+                    const content = await response.text()
```
Comment (eseliger) [inline client/cody-shared/src/sourcegraph-api/completions/browserClient.ts:25]: should we handle errors of response.text() in a catch handler and make sure we still call cb.onError?

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.46 kb) | 0.00% (+0.46 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/cody-web-error-handling/compare-report.html;tab=live_object) for a full comparison between the commits 4651d2a and 128a0b9 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by eseliger — APPROVED** [review]
tried it locally, works well! One thing I noticed is that when the body is empty (this time I got a 502 from frontend restarting) the message is simply empty - should we add something if the response.text didn't yield anything like a "Request failed with status code XXX" or so? 
Screenshot 2023-04-27 at 13 25 40@2x

File: client/cody-shared/src/sourcegraph-api/completions/browserClient.ts (hunk 1 above)
Comment (philipp-spiess) [reply client/cody-shared/src/sourcegraph-api/completions/browserClient.ts:25]: Will do!

**Review by eseliger — APPROVED** [review]

**Comment by taylorsperry** [comment]
Thank you! The limit of 2 requests per hour isn't right though, is it? I know we're still figuring out what the limits need to be exactly, so is this just a placeholder that we will update once we have aligned on those limits?

**Comment by eseliger** [comment]
no that was just for testing. 50 as agreed upon is configured on dotcom :)

**Pull request merged**